python/rpdk/python/codegen.py

```diff
@@ -281,6 +298,19 @@ def _docker_build(cls, external_path):
             "image '%s' needs to be pulled first.",
             image,
         )
+
+        # Docker will mount the path specified in the volumes variable in the container
+        # and pip will place all the dependent packages inside the volumes/build path.
+        # codegen will need access to this directory during package()
+        # Try to get current effective user ID and Group ID. Valid on UNIX-like systems
+        try:
```

Comment:
Is there a way to detect this without a try? Would be better if we didn't rely on an exception for making the decision.

Reply:
There was at least one report of geteuid not working on a new version of MacOS so checking for OS doesn't look to guarantee that this function exists on an OS.  While catching an exception isn't ideal I thought that seemed the most future proof, e.g. when python supports that function on that OS our code doesn't have to change.

Reply:
Updated to check with windows before attempting geteuid() for the edge case of the call still failing I added a LOG.warning that it is defaulting to root:root